eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py

```diff
@@ -70,218 +69,213 @@ def create_table(private_config_xml_file):
-        for assembly in get_all_results_for_query(pg_conn, query):
-            assembly_list.add(assembly[0])
-    return list(assembly_list)
-
-
-def get_filter_group_count_pipeline(taxonomy, assembly_list):
-    filter_by_taxonomy_and_assembly = {
-        "$match":
-            {
-                "tax": {
-                    "$eq": taxonomy,
-                },
-                "asm": {
-                    "$in": assembly_list,
-                }
-            }
-    }
-    group_by_assembly = {
-        "$group":
-            {
-                "_id": {"assembly": "$asm"},
-                "rs_count": {"$sum": 1},
-            }
-    }
-    return [filter_by_taxonomy_and_assembly, group_by_assembly]
-
-
-def get_rs_count_per_assembly_in_collection(mongo_source, coll_name, pipeline):
-    coll = mongo_source.mongo_handle[mongo_source.db_name][coll_name]
-    agg_result = coll.aggregate(pipeline, batchSize=0)
-    agg_res_dict = {}
-    for entry in agg_result:
-        agg_res_dict[entry['_id']['assembly']] = entry['rs_count']
-    return agg_res_dict
-
-
-def get_sources_and_rs_count(cve_res, dbsnp_res, assembly):
-    if assembly in cve_res and assembly in dbsnp_res:
-        return ['DBSNP, EVA', cve_res[assembly] + dbsnp_res[assembly]]
-    elif assembly in cve_res and assembly not in dbsnp_res:
-        return ['EVA', cve_res[assembly]]
-    elif assembly not in cve_res and assembly in dbsnp_res:
-        return ['DBSNP', dbsnp_res[assembly]]
+        for tax, asm_acc in get_all_results_for_query(pg_conn, query):
+            insert_entry_for_taxonomy_and_assembly(private_config_xml_file, profile, tax, asm_acc, release_version,
+                                                   ref_dir)
+
+
+def insert_entry_for_taxonomy_and_assembly(private_config_xml_file, profile, tax, asm_acc, release_version, ref_dir):
+    with get_metadata_connection_handle(profile, private_config_xml_file) as pg_conn:
+        sc_name = get_scientific_name(pg_conn, tax)
+        src = 'EVA'
+        ncbi_assembly = NCBIAssembly(asm_acc, sc_name, ref_dir)
+        fasta_path = ncbi_assembly.assembly_fasta_path
+        report_path = ncbi_assembly.assembly_report_path
+        tempongo_instance = get_tempmongo_instance(pg_conn, tax)
+        release_folder_name = normalise_taxon_scientific_name(sc_name)
+
+        insert_query = f"""INSERT INTO eva_progress_tracker.clustering_release_tracker(
+                            taxonomy, scientific_name, assembly_accession, release_version, sources,
+                            fasta_path, report_path, tempmongo_instance, release_folder_name) 
+                            VALUES ({tax}, '{sc_name}', '{asm_acc}', {release_version}, '{src}', 
+                            '{fasta_path}', '{report_path}', '{tempongo_instance}', '{release_folder_name}') 
+                            ON CONFLICT DO NOTHING"""
```

Comment:
If you try to insert a taxonomy/assembly that is already in the database for an additional source this will not work. 
For example if we load a tax1/asm1 DBSNP row and we subsequently want to load tax1/asm1 EVA, then we won't be able to.

Reply:
updated